Missing URL Protocol Scheme check in newer versions - now provider is crashing

### Terraform Version
* Terraform: `Terraform v1.0.11 on darwin_amd64`
* Terraform Grafana Provider: `1.16.0`
* Grafana: `v8.2.2 on AWS (AMG)`

### Affected Resource(s)
Please list the resources as a list, for example:
* grafana_data_source

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
provider "grafana" {
  url  = format("%s.grafana-workspace.eu-west-1.amazonaws.com/", var.grafana_workspace_id)
  auth = var.grafana_api_key
}

resource "grafana_data_source" "prometheus" {
  for_each = { for datasource in var.datasources : datasource.name => datasource }

  type       = "prometheus"
  name       = each.value.name
  is_default = each.value.is_default
  url        = each.value.amp_url
  json_data {
    http_method     = "GET"
    sigv4_auth      = true
    sigv4_assume_role_arn = each.value.role_arn
    sigv4_auth_type = "workspace-iam-role"
    sigv4_region    = "eu-west-1"
  }
  depends_on = [
    aws_iam_role.grafana_assume_read_role
  ]
}
```
**Example *.tfvars file:**
``` 

datasources = [
  {
    name       = "Dev",
    amp_url    = "https://myprometheusinstance",
    role_arn   = "arn:aws:iam::123456789:role/read"
    is_default = false
  },
  {
    name       = "Staging",
    amp_url    = "https://myprometheusinstance2",
    role_arn   = "arn:aws:iam::987654321:role/read"
    is_default = true
  }
]
```

### Debug Output


### Expected Behavior
On previous versions (last tested with 1.13.0), the provider printed an error stating something like  `protocol scheme "" is not supported` and aborted the process, which lead me to the correct issue. 

This error is currently only visible in the debugging logs: 
``2021-11-30T09:15:23.010+0100 [INFO]  provider.terraform-provider-grafana_v1.16.0: 2021/11/30 09:15:23 [ERROR] Grafana API Request error: &errors.errorString{s:"unsupported protocol scheme \"\""}: timestamp=2021-11-30T09:15:23.009+0100``

### Actual Behavior
On newer versions the script simply fails with the following error. No validity check against the URL is done. 
The script simply crashes the following output:
```
grafana_data_source.prometheus["Dev"]: Still creating... [10s elapsed]
grafana_data_source.prometheus["Staging"]: Still creating... [10s elapsed]
2021-11-30T09:15:18.006+0100 [INFO]  provider.terraform-provider-grafana_v1.16.0: 2021/11/30 09:15:18 [ERROR] Grafana API Request error: &errors.errorString{s:"unsupported protocol scheme \"\""}: timestamp=2021-11-30T09:15:18.006+0100
2021-11-30T09:15:18.006+0100 [INFO]  provider.terraform-provider-grafana_v1.16.0: 2021/11/30 09:15:18 [ERROR] Grafana API Request error: &errors.errorString{s:"unsupported protocol scheme \"\""}: timestamp=2021-11-30T09:15:18.006+0100
...
Stack trace from the terraform-provider-grafana_v1.16.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x178594b]

goroutine 66 [running]:
github.com/grafana/grafana-api-golang-client.(*Client).request(0xc00054c8f0, 0x19c8643, 0x4, 0x19d15f1, 0x10, 0x0, 0x1acba00, 0xc0004f8090, 0x1881b20, 0xc0005e4000, ...)
        github.com/grafana/grafana-api-golang-client@v0.1.0/client.go:114 +0x26b
github.com/grafana/grafana-api-golang-client.(*Client).NewDataSource(0xc00054c8f0, 0xc000640000, 0x0, 0x0, 0xc000620600)
        github.com/grafana/grafana-api-golang-client@v0.1.0/datasource.go:131 +0x145
github.com/grafana/terraform-provider-grafana/grafana.CreateDataSource(0x1adedc8, 0xc00049aa20, 0xc0006ae800, 0x186ffa0, 0xc0004910d0, 0xc0000a6870, 0x12e736a, 0xc0006ae680)
        github.com/grafana/terraform-provider-grafana/grafana/resource_data_source.go:388 +0x105
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0003668c0, 0x1aded58, 0xc000620600, 0xc0006ae800, 0x186ffa0, 0xc0004910d0, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:341 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0003668c0, 0x1aded58, 0xc000620600, 0xc00054c750, 0xc0006ae680, 0x186ffa0, 0xc0004910d0, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:467 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000c0a8, 0x1aded58, 0xc000620600, 0xc0004a4050, 0xc000620600, 0x198cae0, 0xc000590a00)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/grpc_provider.go:977 +0xacf
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00013f840, 0x1adee00, 0xc000620600, 0xc0005f8000, 0xc00013f840, 0xc000590a20, 0xc000401ba0)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x198cae0, 0xc00013f840, 0x1adee00, 0xc000590a20, 0xc00049a180, 0x0, 0x1adee00, 0xc000590a20, 0xc000574000, 0x7c8)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000318540, 0x1ae6618, 0xc000113500, 0xc000238240, 0xc0000aa6c0, 0x1f911a0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.42.0/server.go:1282 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000318540, 0x1ae6618, 0xc000113500, 0xc000238240, 0x0)
        google.golang.org/grpc@v1.42.0/server.go:1616 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000130210, 0xc000318540, 0x1ae6618, 0xc000113500, 0xc000238240)
        google.golang.org/grpc@v1.42.0/server.go:921 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.42.0/server.go:919 +0x1fd

Error: The terraform-provider-grafana_v1.16.0 plugin crashed!
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. remove the "https://" prefix in the provider configuration
2. `terraform apply`

### Important Factoids
Simply the "https://" prefix is missing, which causes the issue.

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/grafana/terraform-provider-grafana
 go 1.16
 
 require (
-	github.com/grafana/grafana-api-golang-client v0.1.2
+	github.com/grafana/grafana-api-golang-client v0.1.3
 	github.com/grafana/synthetic-monitoring-agent v0.4.1
 	github.com/grafana/synthetic-monitoring-api-go-client v0.3.0
 	github.com/hashicorp/go-cleanhttp v0.5.2

